### PR TITLE
Added function import_logfile

### PR DESCRIPTION
**Pull request by dspreitz**
Please review & comment.

### CODE REVIEW

**Comment by Meisterschueler** [comment]
Hello Dominic, thank you for the pull request. Please
1. fix the flake8 errors
2. update the CONTRIBUTORS and the README.md

**Comment by dspreitz** [comment]
Do I need to fix the tests too?
If yes, need help on this one.

**Comment by kerel-fs** [comment]
You could add a short logfile example in [tests/](https://github.com/Meisterschueler/ogn-python/tree/master/tests) and call `import_logfiles` from a new test function in [tests/gateway/manage.py](https://github.com/Meisterschueler/ogn-python/blob/master/tests/gateway/test_manage.py).

File: ogn/gateway/manage.py [hunk 1]
```diff
@@ -40,3 +40,28 @@ def run(aprs_user='anon-dev', logfile='main.log', loglevel='INFO'):
 
     client.disconnect()
     logging.shutdown()
+
+
+@manager.command
+def import_logfiles(ogn_logfile, logfile='main.log', loglevel='INFO'):
```
Comment (kerel-fs) [inline ogn/gateway/manage.py]: Imho `import_logfile` is a better name for this function.

Comment (dspreitz) [inline ogn/gateway/manage.py]: Yes, agree, but I wanted to distinguish to the logfile generated by the app.
So how about import_ogn_logfile.

**Comment by dspreitz** [comment]
Due to lack of date information the change is presently not working.
```
2016-09-22 09:26:38,135 - ERRO - ogn.gateway.process - Drop packet, Can't reconstruct timstamp, -4553s from past.
2016-09-22 09:26:38,136 - ERRO - ogn.gateway.process - Received message: ICA3FF44B>APRS,qAS,Walbeck:/104231h5117.99N\00621.37E^347/101/A=001023 !W38! id213FF44B -118fpm -0.8rot 6.5dB 1e -8.9kHz gps1x1
```

**Comment by dspreitz** [comment]
@kerel-fs: I am unfamiliar with Travis CI and copy-pasting with minor mods did not do the job :-( to create a new test function for import_logfile. Can you help?

**Comment by Meisterschueler** [comment]
@dspreitz:
1. you can install flake8 and check your code before you commit it
2. the aprs messages lack of the date. The framework "python-ogn-client" ("ogn-python" is based on it) uses a "reference time" for calculating a plausible timestamp (createTimestamp in python-ogn-client/ogn/parser/utils.py). If you want to import aprs messages of a specific day this function should be altered... @kerel-fs ?

**Comment by kerel-fs** [comment]
Since a logfile doesn't contain a date, it is impossible to import it without further information provided.
As @Meisterschueler recommended, you could provide the creation date of the logfile via an commandline argument `date`. Note: With this approach it wouldn't be possible to import a logfile which spans multiple days.

-

The development tools (flake8, nosetest) can be installed by
```
pip install -e . [dev]
```

File: tests/gateway/test_manage.py [hunk 2]
```diff
@@ -18,5 +19,18 @@ def test_run_user_interruption(self, mock_aprs_client):
         instance.disconnect.assert_called_once_with()
 
 
+class GatewayManagerTest(unittest.TestCase):
+    # try to import stored OGN logfile
+    @mock.patch('ogn.gateway.manage.import_logfile')
+    def test_run_import_logfile(self, mock_import_logfile)
```
Comment (Meisterschueler) [inline tests/gateway/test_manage.py]: ":" missed at the end of the line...

**Comment by kerel-fs** [comment]
This import will fail for all fixes older or newer than 6h from the `reference_date` datetime.

To reconstruct the datetime of a received fix, the parser tries to guess the date of the the fix not only from the reference date but from also from the reference time.
This makes the system reliable against network lag (up to 12h). Everything else is dropped and produces the error message you reported via IRC.

We should rename `reference_date` to `reference_datetime` (this would require a version bump of *python-ogn-client*).

**Pull request merged**